src/simple_paths.rs

```diff
@@ -0,0 +1,165 @@
+use std::{
+    hash::Hash,
+    iter::{from_fn, FromIterator},
+};
+
+use indexmap::IndexSet;
+
+use crate::{
+    Direction::Outgoing,
+    visit::{
+        GraphBase,
+        IntoNeighborsDirected,
+        NodeCount,
+    },
+};
+
+/// Returns iterator that produces all simple paths from `from` node to `to`, which contains at least `min_intermidiate_nodes` nodes
+/// and at most `max_intermidiate_nodes`, if given, limited by graph's order otherwise
+/// Simple path is path without repetitions
+/// Algorithm is adopted from https://networkx.github.io/documentation/stable/reference/algorithms/generated/networkx.algorithms.simple_paths.all_simple_paths.html
+pub fn all_simple_paths<'g, TargetColl: 'g, G>(graph: &'g G,
```

Comment:
Please substitute &'g G with plain G. Everywhere -- it's an amazing trick and means that `&g` and `Reversed(&g)` will both work, for example.

See other algo for examples.